File: twoPointers/11_ContainerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        if (height.length == 0) return 0;
        
        int maximumArea = 0;
        int l = 0, r = height.length - 1;
        while (l < r) {
            maximumArea = Math.max(maximumArea, Math.min(height[l], height[r]) * (r - l));
            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return maximumArea;
    }
}

File: twoPointers/11_ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        result = 0
        while left < right:
            area = min(height[left], height[right]) * (right - left)
            result = max(area, result)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return result
# Need to look at the formal proof for this problem.